ibis/pyspark/tests/test_timecontext.py

```diff
@@ -13,3 +15,172 @@ def test_table_with_timecontext(client):
     expected = table.compile().toPandas()
     expected = expected[expected.time.between(*context)]
     tm.assert_frame_equal(result, expected)
+
+
+def test_window_with_timecontext(client):
+    table = client.table('time_indexed_table')
+    context = (
+        pd.Timestamp('20170102 07:00:00', tz='UTC'),
+        pd.Timestamp('20170103', tz='UTC'),
+    )
+    window1 = ibis.trailing_window(
+        preceding=ibis.interval(hours=1), order_by='time', group_by='key'
+    )
+    window2 = ibis.trailing_window(
+        preceding=ibis.interval(hours=2), order_by='time', group_by='key'
+    )
+    result = table.mutate(
+        mean_1h=table['value'].mean().over(window1),
+        mean_2h=table['value'].mean().over(window2),
+    ).compile(timecontext=context)
+    result_pd = result.toPandas()
+    # load the entire table
+    df = table.compile().toPandas()
+    expected_win_1 = (
+        df.set_index('time')
+        .groupby('key')
+        .value.rolling('1h', closed='both')
+        .mean()
+        .rename('mean_1h')
+    )
+    expected_win_2 = (
+        df.set_index('time')
+        .groupby('key')
+        .value.rolling('2h', closed='both')
+        .mean()
+        .rename('mean_2h')
+    )
+    df = df.set_index('time')
+    df = df.assign(
+        mean_1h=expected_win_1.sort_index(level=['time', 'key']).reset_index(
```

Comment:
can you break this test up, its huge (make a fixture if needed)

Reply:
parameterize is even better if you can

Reply:
Yeah I mark this test xfail for https://github.com/ibis-project/ibis/issues/2453. Let me revisit this in follow up for #2453